expander.go

```diff
@@ -607,25 +608,35 @@ func shouldStopOnError(err error, opts *ExpandOptions) bool {
 	return false
 }
 
-// ExpandSchema expands the refs in the schema object with reference to the root object
-// go-openapi/validate uses this function
-// notice that it is impossible to reference a json scema in a different file other than root
-func ExpandSchema(schema *Schema, root interface{}, cache ResolutionCache) error {
-	// Only save the root to a tmp file if it isn't nil.
-	var base string
+// baseForRoot loads in the cache the root document and produces a fake "root" base path entry
+// for further $ref resolution
+func baseForRoot(root interface{}, cache ResolutionCache) string {
+	// cache the root document to resolve $ref's
+	const rootBase = "root"
 	if root != nil {
-		base, _ = absPath("root")
+		base, _ := absPath(rootBase)
+		normalizedBase := normalizeAbsPath(base)
```

Comment:
`absPath` returns an absolute filesystem path, and then `normalizeAbsPath` normalizes it as a URL path?  That seems strange.  What does that do?

Reply:
I did not really change the behavior here.
This stuff with normalizing URIs is hairy

Reply:
my essential point here was to adopt the same behavior in ExpandResponse() and ExpandParam() as we do for ExpandSchema()

Reply:
And to support ExpandResponse/Params with a root document instead of a file path, to support the validate use case